hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/testutils/HoodieClientTestHarness.java

```diff
@@ -418,4 +445,166 @@ public HoodieTableFileSystemView getHoodieTableFileSystemView(HoodieTableMetaCli
     }
     return Pair.of(partitionPathStatMap, globalStat);
   }
+
+  /**
+   * Validate the metadata tables contents to ensure it matches what is on the file system.
+   */
+  public void validateMetadata(HoodieTestTable testTable, List<String> inflightCommits, HoodieWriteConfig writeConfig, String metadataTableBasePath) throws IOException {
```

Comment:
this is very very long method.

Reply:
Refactored acoording to Siva's suggestions.